proof/refine/ARM/VSpace_R.thy

```diff
@@ -3107,6 +3107,45 @@ lemma isPDCap_PD :
   "isPDCap (ArchObjectCap (PageDirectoryCap r m))"
   by (simp add: isPDCap_def)
 
+lemma lookupIPCBuffer_valid_ipc_buffer [wp]:
+  "\<lbrace>valid_objs'\<rbrace> lookupIPCBuffer b s \<lbrace>case_option \<top> valid_ipc_buffer_ptr'\<rbrace>"
+  unfolding lookupIPCBuffer_def ARM_H.lookupIPCBuffer_def
+  apply (simp add: Let_def getSlotCap_def getThreadBufferSlot_def
+                   locateSlot_conv threadGet_def comp_def)
+  apply (wp getCTE_wp getObject_tcb_wp | wpc)+
+  apply (clarsimp simp del: imp_disjL)
+  apply (drule obj_at_ko_at')
+  apply (clarsimp simp del: imp_disjL)
+  apply (rule_tac x = ko in exI)
+  apply (frule ko_at_cte_ipcbuffer)
+  apply (clarsimp simp: cte_wp_at_ctes_of simp del: imp_disjL)
+  apply (clarsimp simp: valid_ipc_buffer_ptr'_def)
+  apply (frule (1) ko_at_valid_objs')
+   apply (clarsimp simp: projectKO_opts_defs split: kernel_object.split_asm)
+  apply (clarsimp simp add: valid_obj'_def valid_tcb'_def
```

Comment:
You can remove `add` right?

Reply:
If so then fix up indentation in the line directly below this too

Reply:
Okay, I see that the lemmas with `clarsimp simp add:` were just lemmas you moved from later files, not from your new proofs. In that case, no need to change them I think, otherwise it'd go into a cleanup commit.

Reply:
Yes, these lemmas were moved but I'm happy to still do the cleanup here while I'm touching them. @mktnk3 are you ok with that?

Reply:
sure, of course!